frontend/src/commons/ShareDialog/AllContactsItem.js

```diff
@@ -0,0 +1,171 @@
+      .filter(record => !record.viewSwitchReadonly)
+      .some(record => record.canViewEvent)
+    const toggleShare = records
+      .filter(record => !record.shareSwitchReadonly)
+      .some(record => record.canShareEvent)
+
+    setViewChecked(toggleView)
+    setShareChecked(toggleShare)
+  }, [records, setViewChecked, setShareChecked])
+
+  const switchView = useCallback(() => {
+    if (!viewChecked) {
+      setViewChecked(true);
+      const newRecords = records.map((record) => {
+        if (record.viewSwitchReadonly) {
+          return record;
+        }
+
+        addInvitation(record.describes, ['view']);
+        return { ...record, canViewEvent: true }
+      })
+      setRecords(newRecords)
+    } else {
+      setViewChecked(false);
+      setShareChecked(false);
+      const newRecords = records.map((record) => {
+        if (record.viewSwitchReadonly) {
+          return record
+        }
+
+        removeInvitation(record.describes);
+        return { ...record, canViewEvent: false }
+      })
+      setRecords(newRecords)
+    }
+  }, [
+    records,
+    setRecords,
+    addInvitation,
+    removeInvitation,
+    viewChecked,
+    setViewChecked,
+    setShareChecked
+  ]);
+
+  const switchShare = useCallback(() => {
+    if (!shareChecked) {
+      setShareChecked(true);
+
+      const newRecords = records.map((record) => {
+        if (record.shareSwitchReadonly) {
+          return record;
+        }
+
+        if (record.canViewEvent) {
+          addInvitation(record.describes, ['share']);
+          return { ...record, canShareEvent: true }
+        }
+        else {
+          setViewChecked(true);
+          addInvitation(record.describes, ['view', 'share']);
+          return { ...record, canViewEvent: true, canShareEvent: true }
+        }
+      })
+      setRecords(newRecords)
+    } else {
+      setShareChecked(false);
+
+      const newRecords = records.map((record) => {
+        if (record.shareSwitchReadonly) {
+          return record;
+        }
+        if (record.canViewEvent) {
+          removeInvitation(record.describes);
+          return { ...record, canShareEvent: false }
+        } else {
+          addInvitation(record.describes, ['view']);
+          return { ...record, canViewEvent: true, canShareEvent: false }
+        }
+      })
+      setRecords(newRecords)
+    }
+  }, [
+    records,
+    setRecords,
+    addInvitation,
+    removeInvitation,
+    shareChecked,
+    setViewChecked,
+    setShareChecked
+  ]);
+
+  return (
+    <ListItem className={classes.listItem}>
+      <ListItemAvatar className={classes.avatarItem}>
+        <Avatar className={classes.avatar}>
+          C
+        </Avatar>
+
+      </ListItemAvatar>
+      <ListItemText
+        className={classes.primaryText}
+        primary={translate('app.share.all_contacts')}
+      />
+      <ListItemText
+        className={classes.secondaryText}
+        primary={translate('app.share.allow_view')}
```

Comment:
IMO it would be better to use the same wording as other lines. There are two existing keys: `app.permission.view` and `app.permission.share`.